pkg/deploy/api/types.go

```diff
@@ -77,17 +79,20 @@ type RecreateDeploymentStrategyParams struct {
 	// pod is created. All LifecycleHookFailurePolicy values are supported.
 	Mid *LifecycleHook
 	// Post is a lifecycle hook which is executed after the strategy has
-	// finished all deployment logic. The LifecycleHookFailurePolicyAbort policy
-	// is NOT supported.
+	// finished all deployment logic.
 	Post *LifecycleHook
 }
 
-// LifecycleHook defines a specific deployment lifecycle action.
+// LifecycleHook defines a specific deployment lifecycle action. Only one type of action may be specified at any time.
 type LifecycleHook struct {
 	// FailurePolicy specifies what action to take if the hook fails.
 	FailurePolicy LifecycleHookFailurePolicy
+
 	// ExecNewPod specifies the options for a lifecycle hook backed by a pod.
 	ExecNewPod *ExecNewPodHook
+
+	// TagImages instructs the deployer to tag the current image referenced under a container onto an image stream tag if the deployment succeeds.
+	TagImages []TagImageHook
```

Comment:
No, and no.

On Thu, Feb 11, 2016 at 9:21 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/deploy/api/types.go
> <https://github.com/openshift/origin/pull/6925#discussion_r52606256>:
>
> >  	// ExecNewPod specifies the options for a lifecycle hook backed by a pod.
> >  	ExecNewPod *ExecNewPodHook
> > +
> > +	// TagImages instructs the deployer to tag the current image referenced under a container onto an image stream tag if the deployment succeeds.
> > +	TagImages []TagImageHook
>
> If we will not use a different type, we should at least disallow them
> during pre and mid validation. But I still feel this is a different type.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6925/files#r52606256>.
>